packages/dev/parcel-packager-ssg/SSGPackager.js

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const {Packager} = require('@parcel/plugin');
+const {PromiseQueue, urlJoin, blobToString} = require('@parcel/utils');
+const Module = require('module');
+const path = require('path');
+const ReactDOMServer = require('react-dom/server');
+const React = require('react');
+const vm = require('vm');
+const {createBuildCache} = require('@parcel/core/lib/buildCache');
+
+const packagingBundles = createBuildCache();
+const moduleCache = createBuildCache();
+
+module.exports = new Packager({
+  async package({bundle, bundleGraph, getInlineBundleContents}) {
+    let queue = new PromiseQueue({maxConcurrent: 32});
+    bundle.traverse(node => {
+      if (node.type === 'dependency') {
+        let dep = node.value;
+        let entryBundle = bundleGraph.getReferencedBundle(dep, bundle);
+        if (entryBundle?.bundleBehavior === 'inline') {
+          queue.add(async () => {
+            if (!packagingBundles.has(entryBundle)) {
+              packagingBundles.set(entryBundle, getInlineBundleContents(
+                entryBundle,
+                bundleGraph
+              ));
+            }
+
+            let packagedBundle = await packagingBundles.get(entryBundle);
+            let contents = await blobToString(packagedBundle.contents);
+            contents = `module.exports = ${contents}`;
+            return [entryBundle.id, [entryBundle.getMainEntry(), contents]];
+          });
+        }
+      } else if (node.type === 'asset') {
+        let asset = node.value;
+        queue.add(async () => [asset.id, [asset, await asset.getCode()]]);
+      }
+    });
+
+    let assets = new Map(await queue.run());
+    let load = (id) => {
+      const cachedModule = moduleCache.get(id);
+      if (cachedModule) {
+        return cachedModule.exports;
+      }
+
+      let [asset, code] = assets.get(id);
+      let moduleFunction = vm.compileFunction(code, ['exports', 'require', 'module', '__dirname', '__filename'], {
```

Comment:
This simulates a CommonJS environment. I originally tried to use node's `module` to do this by creating synthetic modules, but it had a memory leak so I went to this so we have control over when these modules are evicted from the cache.